### DISCUSSION TITLE
TypeError: undefined is not an object (evaluating 'previous.stack.split')

### DISCUSSION THREAD

**Discussion by Borderliner** [mikro-orm/mikro-orm#5718] [General]
### Describe the bug

While trying to create a user entity and persist it to a sqlite database, I receive this error. Actually when I send a post request on my Elysia endpoint, it creates the requested user from body successfully, but the second time I try to create another user, it fails with this cryptic error. Honestly Mikro-ORM's biggest issue so far has been it's weird error messages which are almost impossible to track down (or I can't, I don't know). The strange thing is it uses "insert into table" the first time I post, but it "updates" the second time. Here's the full error:

```
[query] insert into `users` (`uuid`, `username`, `email`, `display_name`, `password`, `coins`, `created_at`, `updated_at`, `role_id`) values ('188723bf-e577-46d5-af93-2d1a9ae36abf', 'asdafawfqw2af', 'sgswgwE23GEG@gmail.com', 'SWEGFWGWEGHW', '$2b$10$Uz0aqMX6Y9M1m.OQM1tckeM0cW3XUhhIR9S6v8VMQvpoUwtpeKZqK', 0, 1718652767502, 1718652767502, 3) [took 2 ms, 1 row affected]
[query] update `users` set `password` = NULL, `updated_at` = 1718652767507 where `uuid` = 'ea6b71f5-560c-45ca-9470-f133d7cfc68b' [took 2 ms]
[query] rollback
13 |     constructor(previous) {
14 |         super(previous.message);
15 |         Object.getOwnPropertyNames(previous).forEach(k => this[k] = previous[k]);
16 |         this.name = this.constructor.name;
17 |         Error.captureStackTrace(this, this.constructor);
18 |         this.stack += '\n\n' + previous.stack.split('\n').filter(l => l.trim().startsWith('at ')).join('\n');
                                    ^
TypeError: undefined is not an object (evaluating 'previous.stack.split')
      at new DriverException (C:\Users\X\Projects\smm_est\server\node_modules\@mikro-orm\core\exceptions.js:18:32)
      at new ServerException (:1:33)
      at new ConstraintViolationException (:1:33)
      at new NotNullConstraintViolationException (:1:33)
      at convertException (C:\Users\X\Projects\smm_est\server\node_modules\@mikro-orm\sqlite\SqliteExceptionConverter.js:23:20)
      at C:\Users\X\Projects\smm_est\server\node_modules\@mikro-orm\core\drivers\DatabaseDriver.js:348:19
```

### Reproduction

[Reproduction repo](https://github.com/Borderliner/elysia_mikro_server)

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

6.2.9

### Node.js version

Bun v1.1.12 (Windows native)

### Operating system

Windows 11

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I am rather sure this is a bun bug, I believe I already pointed out this in one of the other threads. 

Anyway, if you want me to look into this, I would like to see a minimal reproduction, a simple script, this is way too huge, it's not really a reproduction to begin with.

> Honestly Mikro-ORM's biggest issue so far has been it's weird error messages

Please don't judge MikroORM based on how node-incompatible bun actually is. This is not "weird MikroORM error message", this is bun's weird behavior, like `error.stack` being undefined for whatever reason (most likely some premature optimization of theirs). 

With that said, I will be happy to look into this (once you provide actual minimal reproduction), it could be solvable on our end too - that doesn't change the fact that its bun's issue and we would only work it around from our end.

**Reply by Borderliner** [reply]
@B4nan Understandable. Time to scrap this project I guess. Bun doesn't seem to be ready for production.

**Reply by Borderliner** [reply]
Anyway I found the solution. The culprit was that `password` column in User entity was optional and it was defined like:
```
public password?: string
```
Making it mandatory by removing the question mark fixed the problem.

**Comment by chneau** [comment]
As pointed out by this answer https://github.com/oven-sh/bun/issues/4746#issuecomment-1785336739 you can work around this by adding `Error.prototype.stack = "";` early in your code.
